packages/runtime/src/schema.ts

```diff
@@ -709,22 +802,22 @@ class SchemaBuilder {
     return field.required ? new GraphQLNonNull(type) : type
   }
 
-  _buildFiltersType(objectName: string, fields: RuntimeObjectFields) {
+  _buildFiltersType(objectName: string, fields: RuntimeObjectFields, isInterface = false) {
     const objectInputName = `${objectName}ObjectFilter`
     const inputName = `${objectName}Filters`
 
     const config: GraphQLInputFieldConfigMap = {}
     for (const [key, field] of Object.entries(fields)) {
       let type: GraphQLInputObjectType | undefined
-      if (field.type === 'reference' && (field as RuntimeReference).indexed) {
+      if (field.type === 'reference' && ((field as RuntimeReference).indexed || isInterface)) {
         if (field.refType === 'enum') {
           const enumType = this.#types[field.refName] as GraphQLEnumType
           type = createEnumValueFilterInput(enumType)
           this.#inputObjects[type.name] = type
         } else if (field.refType === 'node') {
```

Comment:
most interfaces will be type node right?

Reply:
Yes, `node` for a reference to a single document and `connection` for many